### PR TITLE
OWManifoldLearning: Clean up UX

### LINKED ISSUE #5248: Clean up Manifold Learner UX
- Use QFormLayout,
- probably replace spins with line edits,
- double colons...

### PR DESCRIPTION
**Pull request by janezd**
### Issue

I refreshed Manifold Learning as discussed on Tuesday on @irgolic's *Widgets showing show*.

Closes #5248.

### Description of changes

- Use `QFormLayout` so that controls are properly aligned
- ~~Replace spins with line edits~~
- Put parameters in the same box as the combo for selecting the method. These are method's parameters, and they even change when the method is changed, so I think they must be in the same box.

Screenshot 2021-02-11 at 19 24 59

### Considerations

- ~~I'm having second thoughts about removing the (now unnecessary) `_create_spin_parameter`. Do we keep it, just for the case we need it in the future? The method for check boxes is also still there.~~
- ~~Number of components is still a spin, because it probably makes sense. Keep it like this or turn it into line edit, too?~~
- I haven't tried whether the widget still works. :) It seems to pass the tests, though.

##### Includes
- [X] Code changes

### CODE REVIEW

**Comment by ajdapretnar** [comment]
Locally Linear Embedding looks a bit off, but the rest are ok.

**Review by VesnaT — CHANGES_REQUESTED** [review]
The `QIntValidator` does not suffice and I'm getting various error typing into the line edits.

File: Orange/widgets/unsupervised/owmanifoldlearning.py [hunk 1]
```diff
@@ -114,14 +123,14 @@ def __init__(self, parent):
...
             "metric", "Metric:")
-        self.perplexity_spin = self._create_spin_parameter(
-            "perplexity", 1, 100, "Perplexity:")
-        self.early_exaggeration_spin = self._create_spin_parameter(
-            "early_exaggeration", 1, 100, "Early exaggeration:")
-        self.lr_spin = self._create_spin_parameter(
-            "learning_rate", 1, 1000, "Learning rate:")
-        self.n_iter_spin = self._create_spin_parameter(
-            "n_iter", 250, 1e5, "Max iterations:")
+        self.perplexity_spin = self._create_line_edit_parameter(
```
[the remark this answers is not in the document]
Comment (janezd) [reply Orange/widgets/unsupervised/owmanifoldlearning.py]: I wonder who introduced those variables in the first place. Who's to `blame`? :)

You're right, though. I removed them in the second commit.

**Comment by janezd** [comment]
I scaled this back. `QIntValidator` sucks, so I went back to spins, but in form layout and fixed width.

Except for removing unnecessary variables, this is now only about layout.

**Review by VesnaT — APPROVED** [review]

**Pull request merged**